### DISCUSSION TITLE
Foam modeling in ABAQUS inp

### DISCUSSION THREAD

**Discussion by Jeongdae11** [neperfepx/neper#477] [Q&A] [answered]
Hello, I'm using Neper version 4.4.0 to build microstructure of foam structure. 
To do it, I am generating tessellation and meshing with the following commands :

export OMP_NUM_THREADS=1
neper -T-n from_morpho -morpho “diameq:lognoermal(0.1,0.03),1-sphericity:lognormal(0.145,0.03)”
neper -V nfrom_morpho-id1.tess -print Image_1
neper -M nfrom_morpho-id1.tess -elttype quad -format inp

with that commands, i opened inp file as follows:

   
![ABAQUS inp](https://user-images.githubusercontent.com/105771843/168989493-a6374d9f-84fb-48f7-85ea-1f41acff081a.jpg)

However, what i want to make is as follows: 

![Figure from International Journal of Engineering Science](https://user-images.githubusercontent.com/105771843/168989793-e4d90ddb-84ec-480e-8023-09deae3a8d50.jpg)

Reflecting characteristic of foam structure, I want to make pore structure.  However, I am beginner of Neper and need help....

Thank you very much for your help!

**Comment by rquey** [comment, marked as answer]
You have to mesh into tet elements (the default) instead of quad elements:

```
 neper -T -n 500 -reg 1
 neper -M n500-id1.tess -dim 2 -rcl 0.6
 neper -V n500-id1.tess -datacelltrs 0.5 -print img1
 neper -V n500-id1.tess,n500-id1.msh -showelset2d "domtype!=2" -dataelset2dcol lightblue -dataelt2dedgerad 0.002 -print img2
```
![image](https://user-images.githubusercontent.com/2659376/169022525-ee6f6329-fac8-4a34-b080-b75747156206.png)
![image](https://user-images.githubusercontent.com/2659376/169022536-85b2dd65-55ba-42ec-b9ab-14637520ca2a.png)

**Reply by Jeongdae11** [reply]
Thank you for your helpful comments! it worked! However, inp file that i exported contains outside plane as follows... inside cells(pore) with face are well organized. 

To realize real cellular structure, i should delete outside 6 planes.. Please tell me how to delete if you know
![Inp file 1](https://user-images.githubusercontent.com/105771843/169509719-2f24c0af-60a9-4855-8775-b199dd3fd89a.jpg)
![Inp file 2](https://user-images.githubusercontent.com/105771843/169509731-5430eac0-fc77-48c9-bafa-0f0057b7c1af.jpg)

**Reply by rquey** [reply]
```
neper -M n500-id1.tess -dim 2 -rcl 0.6 -meshface "domtype!=2" -nset none -faset none
```

**Reply by ap264** [reply]
I have question here.

here in -M module, you are using -dim 2. I would like to know the purpose of this. If i can take a guess, does it help the user in generating 2d mesh elements?

**Reply by ap264** [reply]
Hi 
If you don't mind, i have another question.

After using this type of mesh, I have exported it to Abaqus. After running an analysis in abaqus, i get an error saying some elements have zero, small or negative area. the total number of these elements are 24375. Can you help me rectify this error? 

Thank you

**Reply by rquey** [reply]
> I have question here.
> 
> here in -M module, you are using -dim 2. I would like to know the purpose of this. If i can take a guess, does it help the user in generating 2d mesh elements?

Yes, `-dim 2` asks for a 2D mesh.

> After using this type of mesh, I have exported it to Abaqus. After running an analysis in abaqus, i get an error saying some elements have zero, small or negative area. the total number of these elements are 24375. Can you help me rectify this error?

I have no real clue about this. Some elements may have a small area, but none have a zero or a negative area...

**Reply by sg88888** [reply]
Hello, how did you delete only one face

**Reply by sg88888** [reply]
Hello doctor, I want to generate a 3D mesh, I changed DIM 2 to DIM 3, why can't I run it, how can I modify it
